agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go

```diff
@@ -0,0 +1,663 @@
+	ssmParam := filepath.Base(parsedARN.Resource)
+	ssmParams := []string{ssmParam}
+
+	ssmParamMap, err := ssm.GetParametersFromSSM(ssmParams, ssmClient)
+	if err != nil {
+		fv.setTerminalReason(err.Error())
+		return err
+	}
+
+	ssmParamData, _ := ssmParamMap[ssmParam]
+	creds := FSxWindowsFileServerCredentials{}
+
+	if err := json.Unmarshal([]byte(ssmParamData), &creds); err != nil {
+		return err
+	}
+
+	fv.SetCredentials(creds)
+	return nil
+}
+
+func (fv *FSxWindowsFileServerResource) retrieveASMCredentials(credentialsParameterARN string, iamCredentials credentials.IAMRoleCredentials) error {
+	_, err := arn.Parse(credentialsParameterARN)
+	if err != nil {
+		fv.setTerminalReason(err.Error())
+		return err
+	}
+
+	asmClient := fv.asmClientCreator.NewASMClient(fv.region, iamCredentials)
+	asmData, err := asm.GetSecretFromASM(credentialsParameterARN, asmClient)
+	if err != nil {
+		return err
+	}
+
+	creds := FSxWindowsFileServerCredentials{}
+	if err := json.Unmarshal([]byte(asmData), &creds); err != nil {
+		return err
+	}
+
+	fv.SetCredentials(creds)
+	return nil
+}
+
+func (fv *FSxWindowsFileServerResource) retrieveFileSystemDNSName(fileSystemId string, iamCredentials credentials.IAMRoleCredentials) error {
+	fileSystemIds := []string{fileSystemId}
+	fsxClient := fv.fsxClientCreator.NewFSxClient(fv.region, iamCredentials)
+	fileSystemDNSMap, err := fsx.GetFileSystemDNSNames(fileSystemIds, fsxClient)
+	if err != nil {
+		fv.setTerminalReason(err.Error())
+		return err
+	}
+
+	fv.SetFileSystemDNSName(fileSystemDNSMap[fileSystemId])
+	return nil
+}
+
+var execCommand = exec.Command
+
+func (fv *FSxWindowsFileServerResource) performHostMount(remotePath string, localPath string, username string, password string) error {
+	// formatting to keep powershell happy
+	localPath = fmt.Sprintf("-LocalPath %s", localPath)
+	remotePath = fmt.Sprintf("-RemotePath %s", remotePath)
+	creds := fmt.Sprintf("-Credential $(New-Object System.Management.Automation.PSCredential(%s, $(ConvertTo-SecureString %s -AsPlainText -Force)))", username, password)
+
+	cmd := execCommand("powershell.exe",
+		"New-SmbGlobalMapping",
+		localPath,
+		remotePath,
+		creds,
+		"-Persistent $true",
+		"-RequirePrivacy $true",
+		"-ErrorAction Stop")
+
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		fv.setTerminalReason(err.Error())
+		return err
+	}
+
+	cmd = execCommand("powershell.exe",
+		"Get-SmbGlobalMapping",
+		localPath)
+
+	_, err = cmd.CombinedOutput()
+	if err != nil {
+		return errors.New("could not mount fsxwindowsfileserver resource on the container instance")
+	}
+
+	return nil
+}
+
+func (fv *FSxWindowsFileServerResource) handleRootDirectory(rootDirectory string) {
+	dir := utils.GetCanonicalPath(rootDirectory)
+	dir = strings.Trim(dir, "\\")
+
+	fv.SetRootDirectory(dir)
+	return
+}
+
+func (fv *FSxWindowsFileServerResource) Cleanup() error {
+	hostPath := fv.VolumeConfig.FSxWindowsFileServerHostPath
+	fv.clearFSxWindowsFileServerResource(hostPath)
```

Comment:
This always returns `nil` even if the cleanup fails. Is that acceptable?

Reply:
this is to accommodate the case where customer might tamper with the drive mapping on the instance after task is stopped. thats why the warning instead of an error. (just like here - https://github.com/aws/amazon-ecs-agent/blob/master/agent/taskresource/credentialspec/credentialspec_windows.go#L563)